File: frontend/IncidentResponse/src/app/models/Incident.ts
import { PriorityEnum } from "../enums/PriorityEnum";
import { StatusIncidentEnum } from "../enums/StatutsIncidentEnum";
import { Commentary } from "./Commentary";

export class Incident {
    id !: number;
    title !:string;
    description !: string;
    priority !: PriorityEnum;
    status !: StatusIncidentEnum;
    user ?: number ;
    commentary ?:Commentary[]; 
    date !: Date
}

File: backend/src/main/java/com/incidentresponse/domain/model/Incident.java
package com.incidentresponse.domain.model;

import com.incidentresponse.enums.PriorityStatusEnum;
import com.incidentresponse.enums.StatusIncidentEnum;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Date;

@NoArgsConstructor
@Getter
@Setter
public class Incident {


    private Long id;
    private String title;
    private User userAssignated;
    private String description;
    private PriorityStatusEnum priority;
    private StatusIncidentEnum status;
    private Date date;
}
